Objective-C/Tests/DateTimeQueryFunctionTest.m

```diff
@@ -0,0 +1,225 @@
+                          result: [NSNumber numberWithDouble: 1540319581000]];
+    
+    [self validateStringToMillis: @"2018-10-23T18:33:01Z"
+                          result: [NSNumber numberWithDouble: 1540319581000]];
+    
+    [self validateStringToMillis: @"2018-10-23T18:33:01.123Z"
+                          result: [NSNumber numberWithDouble: 1540319581123]];
+    
+    // leap year
+    [self validateStringToMillis: @"2020-02-29T23:59:59.000000+0000"
+                          result: [NSNumber numberWithDouble: 1583020799000]];
+}
+
+
+- (void) testStringToUTC {
+    [self validateStringToUTC: nil result: nil];
+    [self validateStringToUTC: @"x" result: nil];
+    
+    [self validateStringToUTC: @"2018-10-23T18:33:01Z"
+                       result: @"2018-10-23T18:33:01Z"];
+    
+    [self validateStringToUTC: @"2018-10-23T11:33:01-0700"
+                       result: @"2018-10-23T18:33:01Z"];
+    
+    [self validateStringToUTC: @"2018-10-23T11:33:01+03:30"
+                       result: @"2018-10-23T08:03:01Z"];
+    
+    [self validateStringToUTC: @"2018-10-23T18:33:01.123Z"
+                       result: @"2018-10-23T18:33:01.123Z"];
+    
+    [self validateStringToUTC: @"2018-10-23T11:33:01.123-0700"
+                       result: @"2018-10-23T18:33:01.123Z"];
+    
+    [self validateStringToUTC: @"1970-01-01T00:00:00.000000+0000"
+                       result: @"1970-01-01T00:00:00Z"];
+}
+
+
+- (void) testMillisToString {
+    int mSec = 1000;
+    double seconds = 0.0;
+    [self validateMillisToString: [NSNumber numberWithDouble: seconds * mSec]
+                          result: [NSDate dateWithTimeIntervalSince1970: seconds]];
+    
+    seconds = 0.123;
+    [self validateMillisToString: [NSNumber numberWithDouble: seconds * mSec]
+                          result: [NSDate dateWithTimeIntervalSince1970: seconds]];
+    
+    seconds = 1000.123;
+    [self validateMillisToString: [NSNumber numberWithDouble: seconds * mSec]
+                          result: [NSDate dateWithTimeIntervalSince1970: seconds]];
+    
+    seconds = 65789245.123;
+    [self validateMillisToString: [NSNumber numberWithDouble: seconds * mSec]
+                          result: [NSDate dateWithTimeIntervalSince1970: seconds]];
+}
+
+
+- (void) testMillisToUTC {
+    [self validateMillisToUTC: [NSNumber numberWithDouble: 0]
+                       result: @"1970-01-01T00:00:00Z"];
+    
+    [self validateMillisToUTC: [NSNumber numberWithDouble: 1540319581000]
+                       result: @"2018-10-23T18:33:01Z"];
+    
+    [self validateMillisToUTC: [NSNumber numberWithDouble: 1540319581123]
+                       result: @"2018-10-23T18:33:01.123Z"];
+    
+    [self validateMillisToUTC: [NSNumber numberWithDouble: 1540319581999]
+                       result: @"2018-10-23T18:33:01.999Z"];
+}
+
+
+#pragma mark - Helper Methods
+
+
+- (void) validateStringToMillis: (NSString*)input result: (nullable NSNumber*)millis {
```

Comment:
I can make these validate to single function, but that will make things complicated. And pass in an enum or to differentiate between the methods, like string_to_millis, etc. Now a few lines are duplicated.